### DISCUSSION TITLE
Passing Options from module to plugin in Nuxt 3 Modules

### DISCUSSION THREAD

**Discussion by aliyr** [nuxt/nuxt#15980] [Questions]
Hi

I've created a module using [module starter template](https://github.com/nuxt/starter/tree/module).

This is my module code: 

`import { fileURLToPath } from 'url'
import { defineNuxtModule, addPlugin, createResolver } from '@nuxt/kit'

export interface ModuleOptions {
  addPlugin: boolean
}

export default defineNuxtModule<ModuleOptions>({
  meta: {
    name: '@jabama/nuxt-http',
    configKey: 'http',
    compatibility: {
      nuxt: '^3.0.0'
    }
  },
  defaults: {
    addPlugin: true,
    client
  },
  setup (options, nuxt) {
    if (options.addPlugin) {
      const { resolve } = createResolver(import.meta.url)
      const runtimeDir = fileURLToPath(new URL('./runtime', import.meta.url))
      nuxt.options.build.transpile.push(runtimeDir)
      addPlugin({
        src: resolve(runtimeDir, 'plugin'),
        mode: 'all'
      })
    }
  }
})`

and This is my plugin: 

`import { defineNuxtPlugin } from '#app'

export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.provide('injected', () => 'my injected function')
})`

and also This is my nuxt playground config: 

`export default defineNuxtConfig({
  modules: [
    http
  ],
  http: {
    addPlugin: true,
    client: 'axios'
  }
})
`

How can I pass module options to the declared plugin?
for example, passing the "client" option to the plugin

**Comment by wobsoriano** [comment]
https://nuxt.com/docs/api/composables/use-runtime-config

**Reply by aliyr** [reply]
Thank you for your answer
Does it have other options like passing options directly to the specific plugin?
